modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewChangesDisplayContext.java

```diff
@@ -1455,6 +1456,21 @@ else if (_log.isDebugEnabled()) {
 						"groupId", groupedModel.getGroupId());
 				}
 
+				if (model instanceof WorkflowedModel) {
+					int changeType = _ctDisplayRendererRegistry.getChangeType(
+						ctEntry, model);
+
+					if (_ctDisplayRendererRegistry.isWorkflowEnabled(
+							ctEntry.getCompanyId(),
+							modelInfo._jsonObject.getLong("groupId"),
+							_portal.getClassName(
+								ctEntry.getModelClassNameId())) &&
+						(changeType != CTConstants.CT_CHANGE_TYPE_DELETION)) {
```

Comment:
This whole logic should be in the helper method to help reduce duplicate code. The helper should be like this 

`ctDisplayRendererRegistry.isWorkflowEnabled(CTEntry ctEntry, T model)`